web/modules/custom/geolayer/modules/map/src/Plugin/Field/FieldWidget/GeofieldWidget.php

```diff
@@ -67,12 +160,43 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
     $element['value'] = [
       '#type' => 'textarea',
       '#title' => $this->t('Geometry'),
-      '#default_value' => $current_value,
     ];
+    if($current_value) {
+      $element['#default_value'] = $current_value;
+      $element['map']['#map_settings']['wkt'] = $current_value;
+      $element['value']['#default_value'] = $current_value;
+      $element['value']['#value'] = $current_value;
```

Comment:
do we need to set all four values?

Reply:
@peacog we only need to set, ` $element['map']['#map_settings']['wkt']` and ` $element['value']['#value']` here, but i am updating the default values here too, just in case to avoid any unwanted discrepancies in the application state !